src/VisualStudio/Core/Def/ProjectSystem/OpenTextBufferProvider.cs

```diff
@@ -91,9 +99,27 @@ private void RaiseEventForEachListener(Action<IOpenTextBufferEventListener> acti
         }
     }
 
-    private async Task CheckForExistingOpenDocumentsAsync(IThreadingContext threadingContext)
+    private async Task CheckForExistingOpenDocumentsAsync()
     {
-        await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+        _threadingContext.ThrowIfNotOnBackgroundThread();
+
+        /* 
+         * NOTE: UNCOMMENT ONCE https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2441480 IS FIXED
+         *       AND REMOVE THE THREAD CHECK FROM THE INITIALIZATION OF _runningDocumentTable
+
+         // Yield the thread, so the caller can proceed immediately.
+         await Task.Yield();
+
+         // Ensure we obtain the RDT before transitioning to the main thread
+         _ = _runningDocumentTable.Value;
+
+         */
+
+        await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+        /* NOTE: REMOVE ONCE https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2441480 IS FIXED */
+        // Temporarily ensure we obtain the RDT from the main thread
+        _ = _runningDocumentTable.Value;
```

Comment:
You could also just remove this now -- the first use will be inside EnumerateDocumentSet, which it's fine that way.